### ISSUE TITLE
pretrained_model_name_or_path？

### ISSUE THREAD

**Issue by YemoGardner** [Aria-Zhangjl/StoryWeaver#1]
usage: train.py [-h] [--input_perturbation INPUT_PERTURBATION] --pretrained_model_name_or_path PRETRAINED_MODEL_NAME_OR_PATH [--revision REVISION] [--dataset_dir DATASET_DIR] [--data_name DATA_NAME] [--test_dataset_dir TEST_DATASET_DIR]
                [--dataset_config_name DATASET_CONFIG_NAME] [--train_data_dir TRAIN_DATA_DIR] [--image_column IMAGE_COLUMN] [--caption_column CAPTION_COLUMN] [--max_train_samples MAX_TRAIN_SAMPLES] [--validation_prompts VALIDATION_PROMPTS [VALIDATION_PROMPTS ...]]
                [--output_dir OUTPUT_DIR] [--cache_dir CACHE_DIR] [--seed SEED] [--resolution RESOLUTION] [--max_char MAX_CHAR] [--center_crop] [--use_high_res] [--random_flip] [--train_batch_size TRAIN_BATCH_SIZE] [--num_train_epochs NUM_TRAIN_EPOCHS]
                [--max_train_steps MAX_TRAIN_STEPS] [--gradient_accumulation_steps GRADIENT_ACCUMULATION_STEPS] [--gradient_checkpointing] [--learning_rate LEARNING_RATE] [--scale SCALE] [--scale_lr] [--lr_scheduler LR_SCHEDULER]
                [--lr_warmup_steps LR_WARMUP_STEPS] [--snr_gamma SNR_GAMMA] [--dream_detail_preservation DREAM_DETAIL_PRESERVATION] [--use_8bit_adam] [--allow_tf32] [--use_ema] [--non_ema_revision NON_EMA_REVISION] [--dataloader_num_workers DATALOADER_NUM_WORKERS]
                [--adam_beta1 ADAM_BETA1] [--adam_beta2 ADAM_BETA2] [--adam_weight_decay ADAM_WEIGHT_DECAY] [--adam_epsilon ADAM_EPSILON] [--max_grad_norm MAX_GRAD_NORM] [--push_to_hub] [--hub_token HUB_TOKEN] [--prediction_type PREDICTION_TYPE]
                [--hub_model_id HUB_MODEL_ID] [--logging_dir LOGGING_DIR] [--mixed_precision {no,fp16,bf16}] [--report_to REPORT_TO] [--local_rank LOCAL_RANK] [--checkpointing_steps CHECKPOINTING_STEPS] [--checkpoints_total_limit CHECKPOINTS_TOTAL_LIMIT]
                [--resume_from_checkpoint RESUME_FROM_CHECKPOINT] [--enable_xformers_memory_efficient_attention] [--noise_offset NOISE_OFFSET] [--validation_epochs VALIDATION_EPOCHS] [--tracker_project_name TRACKER_PROJECT_NAME]
train.py: error: the following arguments are required: --pretrained_model_name_or_path

May I ask if you can public the model you are using? This way, I can achieve the same results as you. thank you!

**Comment by Aria-Zhangjl** [comment]
> usage: train.py [-h] [--input_perturbation INPUT_PERTURBATION] --pretrained_model_name_or_path PRETRAINED_MODEL_NAME_OR_PATH [--revision REVISION] [--dataset_dir DATASET_DIR] [--data_name DATA_NAME] [--test_dataset_dir TEST_DATASET_DIR] [--dataset_config_name DATASET_CONFIG_NAME] [--train_data_dir TRAIN_DATA_DIR] [--image_column IMAGE_COLUMN] [--caption_column CAPTION_COLUMN] [--max_train_samples MAX_TRAIN_SAMPLES] [--validation_prompts VALIDATION_PROMPTS [VALIDATION_PROMPTS ...]] [--output_dir OUTPUT_DIR] [--cache_dir CACHE_DIR] [--seed SEED] [--resolution RESOLUTION] [--max_char MAX_CHAR] [--center_crop] [--use_high_res] [--random_flip] [--train_batch_size TRAIN_BATCH_SIZE] [--num_train_epochs NUM_TRAIN_EPOCHS] [--max_train_steps MAX_TRAIN_STEPS] [--gradient_accumulation_steps GRADIENT_ACCUMULATION_STEPS] [--gradient_checkpointing] [--learning_rate LEARNING_RATE] [--scale SCALE] [--scale_lr] [--lr_scheduler LR_SCHEDULER] [--lr_warmup_steps LR_WARMUP_STEPS] [--snr_gamma SNR_GAMMA] [--dream_detail_preservation DREAM_DETAIL_PRESERVATION] [--use_8bit_adam] [--allow_tf32] [--use_ema] [--non_ema_revision NON_EMA_REVISION] [--dataloader_num_workers DATALOADER_NUM_WORKERS] [--adam_beta1 ADAM_BETA1] [--adam_beta2 ADAM_BETA2] [--adam_weight_decay ADAM_WEIGHT_DECAY] [--adam_epsilon ADAM_EPSILON] [--max_grad_norm MAX_GRAD_NORM] [--push_to_hub] [--hub_token HUB_TOKEN] [--prediction_type PREDICTION_TYPE] [--hub_model_id HUB_MODEL_ID] [--logging_dir LOGGING_DIR] [--mixed_precision {no,fp16,bf16}] [--report_to REPORT_TO] [--local_rank LOCAL_RANK] [--checkpointing_steps CHECKPOINTING_STEPS] [--checkpoints_total_limit CHECKPOINTS_TOTAL_LIMIT] [--resume_from_checkpoint RESUME_FROM_CHECKPOINT] [--enable_xformers_memory_efficient_attention] [--noise_offset NOISE_OFFSET] [--validation_epochs VALIDATION_EPOCHS] [--tracker_project_name TRACKER_PROJECT_NAME] train.py: error: the following arguments are required: --pretrained_model_name_or_path
> 
> May I ask if you can public the model you are using? This way, I can achieve the same results as you. thank you!

Hi, I use SD 1.5 for my experiments. You can set it up with:
`export MODEL_NAME="runwayml/stable-diffusion-v1-5"`

Additionally, other models in [civitai](https://civitai.com/) can also be used.

**Comment by linjie82** [comment]
> > usage: train.py [-h] [--input_perturbation INPUT_PERTURBATION] --pretrained_model_name_or_path PRETRAINED_MODEL_NAME_OR_PATH [--revision REVISION] [--dataset_dir DATASET_DIR] [--data_name DATA_NAME] [--test_dataset_dir TEST_DATASET_DIR] [--dataset_config_name DATASET_CONFIG_NAME] [--train_data_dir TRAIN_DATA_DIR] [--image_column IMAGE_COLUMN] [--caption_column CAPTION_COLUMN] [--max_train_samples MAX_TRAIN_SAMPLES] [--validation_prompts VALIDATION_PROMPTS [VALIDATION_PROMPTS ...]] [--output_dir OUTPUT_DIR] [--cache_dir CACHE_DIR] [--seed SEED] [--resolution RESOLUTION] [--max_char MAX_CHAR] [--center_crop] [--use_high_res] [--random_flip] [--train_batch_size TRAIN_BATCH_SIZE] [--num_train_epochs NUM_TRAIN_EPOCHS] [--max_train_steps MAX_TRAIN_STEPS] [--gradient_accumulation_steps GRADIENT_ACCUMULATION_STEPS] [--gradient_checkpointing] [--learning_rate LEARNING_RATE] [--scale SCALE] [--scale_lr] [--lr_scheduler LR_SCHEDULER] [--lr_warmup_steps LR_WARMUP_STEPS] [--snr_gamma SNR_GAMMA] [--dream_detail_preservation DREAM_DETAIL_PRESERVATION] [--use_8bit_adam] [--allow_tf32] [--use_ema] [--non_ema_revision NON_EMA_REVISION] [--dataloader_num_workers DATALOADER_NUM_WORKERS] [--adam_beta1 ADAM_BETA1] [--adam_beta2 ADAM_BETA2] [--adam_weight_decay ADAM_WEIGHT_DECAY] [--adam_epsilon ADAM_EPSILON] [--max_grad_norm MAX_GRAD_NORM] [--push_to_hub] [--hub_token HUB_TOKEN] [--prediction_type PREDICTION_TYPE] [--hub_model_id HUB_MODEL_ID] [--logging_dir LOGGING_DIR] [--mixed_precision {no,fp16,bf16}] [--report_to REPORT_TO] [--local_rank LOCAL_RANK] [--checkpointing_steps CHECKPOINTING_STEPS] [--checkpoints_total_limit CHECKPOINTS_TOTAL_LIMIT] [--resume_from_checkpoint RESUME_FROM_CHECKPOINT] [--enable_xformers_memory_efficient_attention] [--noise_offset NOISE_OFFSET] [--validation_epochs VALIDATION_EPOCHS] [--tracker_project_name TRACKER_PROJECT_NAME] train.py: error: the following arguments are required: --pretrained_model_name_or_path
> > May I ask if you can public the model you are using? This way, I can achieve the same results as you. thank you!
> 
> Hi, I use SD 1.5 for my experiments. You can set it up with: `export MODEL_NAME="runwayml/stable-diffusion-v1-5"`
> 
> Additionally, other models in [civitai](https://civitai.com/) can also be used.

是只支持sd1.5吗？

**Comment by Aria-Zhangjl** [comment]
> > > usage: train.py [-h] [--input_perturbation INPUT_PERTURBATION] --pretrained_model_name_or_path PRETRAINED_MODEL_NAME_OR_PATH [--revision REVISION] [--dataset_dir DATASET_DIR] [--data_name DATA_NAME] [--test_dataset_dir TEST_DATASET_DIR] [--dataset_config_name DATASET_CONFIG_NAME] [--train_data_dir TRAIN_DATA_DIR] [--image_column IMAGE_COLUMN] [--caption_column CAPTION_COLUMN] [--max_train_samples MAX_TRAIN_SAMPLES] [--validation_prompts VALIDATION_PROMPTS [VALIDATION_PROMPTS ...]] [--output_dir OUTPUT_DIR] [--cache_dir CACHE_DIR] [--seed SEED] [--resolution RESOLUTION] [--max_char MAX_CHAR] [--center_crop] [--use_high_res] [--random_flip] [--train_batch_size TRAIN_BATCH_SIZE] [--num_train_epochs NUM_TRAIN_EPOCHS] [--max_train_steps MAX_TRAIN_STEPS] [--gradient_accumulation_steps GRADIENT_ACCUMULATION_STEPS] [--gradient_checkpointing] [--learning_rate LEARNING_RATE] [--scale SCALE] [--scale_lr] [--lr_scheduler LR_SCHEDULER] [--lr_warmup_steps LR_WARMUP_STEPS] [--snr_gamma SNR_GAMMA] [--dream_detail_preservation DREAM_DETAIL_PRESERVATION] [--use_8bit_adam] [--allow_tf32] [--use_ema] [--non_ema_revision NON_EMA_REVISION] [--dataloader_num_workers DATALOADER_NUM_WORKERS] [--adam_beta1 ADAM_BETA1] [--adam_beta2 ADAM_BETA2] [--adam_weight_decay ADAM_WEIGHT_DECAY] [--adam_epsilon ADAM_EPSILON] [--max_grad_norm MAX_GRAD_NORM] [--push_to_hub] [--hub_token HUB_TOKEN] [--prediction_type PREDICTION_TYPE] [--hub_model_id HUB_MODEL_ID] [--logging_dir LOGGING_DIR] [--mixed_precision {no,fp16,bf16}] [--report_to REPORT_TO] [--local_rank LOCAL_RANK] [--checkpointing_steps CHECKPOINTING_STEPS] [--checkpoints_total_limit CHECKPOINTS_TOTAL_LIMIT] [--resume_from_checkpoint RESUME_FROM_CHECKPOINT] [--enable_xformers_memory_efficient_attention] [--noise_offset NOISE_OFFSET] [--validation_epochs VALIDATION_EPOCHS] [--tracker_project_name TRACKER_PROJECT_NAME] train.py: error: the following arguments are required: --pretrained_model_name_or_path
> > > May I ask if you can public the model you are using? This way, I can achieve the same results as you. thank you!
> > 
> > 
> > Hi, I use SD 1.5 for my experiments. You can set it up with: `export MODEL_NAME="runwayml/stable-diffusion-v1-5"`
> > Additionally, other models in [civitai](https://civitai.com/) can also be used.
> 
> 是只支持sd1.5吗？


You could also consider using SD 2.1 or other pre-trained models. However, it may be necessary to adjust the training parameters accordingly.
